### PR TITLE
fix: pipe needs to be recreated after stage change (prevent `STOPPED_STAGE_ALTERED`)

### PR DESCRIPTION
**Pull request by joscha**
The pipe will go into [`STOPPED_STAGE_ALTERED`](https://docs.snowflake.com/en/sql-reference/functions/system_pipe_status) status if the underlying stage is modified. This happens for example when using `create_stage = false` on this module and bringing your own.

See:

image

There is no way to put the pipe into `RUNNING` status via a command, the pipe will need to be recreated each time the stage is altered, which is what the changes in this pull request do.

Internal support case reference: https://app.snowflake.com/support/case/00854731

### CODE REVIEW

**Comment by joscha** [comment]
PTAL @sfc-gh-pkommini, @sfc-gh-bkou, @sfc-gh-kumaurya

**Comment by sfc-gh-pkommini** [comment]
lgtm. Thank you for your contribution!

**Comment by sfc-gh-pkommini** [comment]
> lgtm. Thank you for your contribution!

actually sorry. I think same issue. The checks are failing with the version of terraform in the example.
image

**Review by sfc-gh-pkommini — CHANGES_REQUESTED** [review]
Change terraform version in examples so that tests can work.

**Comment by joscha** [comment]
> Change terraform version in examples so that tests can work.

see #23

File: snowpipe.tf [hunk 1]
```diff
@@ -42,4 +42,8 @@ resource "snowflake_pipe" "this" {
   depends_on = [
     snowflake_stage.this
   ]
+
+  replace_triggered_by = [
+    snowflake_stage.this
```
Comment (joscha) [inline snowpipe.tf:47]: is there any way to also recreate when an external stage is passed in, e.g. `create_stage = false`?

**Comment by joscha** [comment]
seems it doesn't like it: `An argument named "replace_triggered_by" is not expected here.`

File: snowpipe.tf (hunk 1 above)
Comment (sfc-gh-pkommini) [reply snowpipe.tf]: I think we can use the data source for stages and check if the stage that was input is present and then if not present recreate the pipe.

Would you be able to contribute a change where a data source: https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/data-sources/stages is fetched and then checked?

Comment (sfc-gh-pkommini) [reply snowpipe.tf]: It may take a coupla weeks for us to get to this.

Comment (joscha) [reply snowpipe.tf]: > Would you be able to contribute a change where a data source: https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/data-sources/stages is fetched and then checked?

can do, could we get the other changes in first and go from there by any chance?

Comment (sfc-gh-pkommini) [reply snowpipe.tf]: 👍🏼 Ofcourse

**Review by sfc-gh-pkommini — APPROVED** [review]
lgtm

**Comment by sfc-gh-pkommini** [comment]
Thank you for putting it into the lifecycle block. :)

**Pull request merged**